build.sh

```diff
@@ -96,34 +96,43 @@ pushd "$BUILD_DIR"
 # The following line installs pyarrow from S3, these wheels have been
 # generated from https://github.com/ray-project/arrow-build from
 # the commit listed in the command.
-if [ -z "$SKIP_PYARROW_INSTALL" ]; then
+if [ -z "$SKIP_THIRDPARTY_INSTALL" ]; then
```

Comment:
This will affect callers... I assume they're external users. What do you think about emitting a warning or error if they have `SKIP_PYARROW_INSTALL` defined but not `SKIP_THIRDPARTY_INSTALL` to let them know? That way it won't silently change behavior. You can check as follows:
```
if [ -n "${SKIP_PYARROW_INSTALL+x}" ] && [ -z "${SKIP_THIRDPARTY_INSTALL+x}" ]; then
  1>&2 echo "SKIP_PYARROW_INSTALL is deprecated; please use SKIP_THIRDPARTY_INSTALL instead."
fi
```

Reply:
Edit: and perhaps fail in that case? Not sure which is better.

Reply:
This isn't a public API (and I'm pretty sure I'm the only one that knows about it).